test/language/expressions/class/private-getter-brand-check-multiple-evaluations-of-class-function-ctor.js

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2019 Jaideep Bhoosreddy (Bloomberg LP). All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Every new evaluation of a class creates a different brand (private getter)
+esid: sec-privatefieldget
+info: |
+  ClassTail : ClassHeritage { ClassBody }
+    ...
+    11. Let proto be ObjectCreate(protoParent).
+    ...
+    31. If PrivateBoundIdentifiers of ClassBody contains a Private Name P such that the P's [[Kind]] field is either "method" or "accessor",
+      a. Set F.[[PrivateBrand]] to proto.
+    ...
+
+  PrivateBrandCheck(O, P)
+    1. If O.[[PrivateBrands]] does not contain an entry e such that SameValue(e, P.[[Brand]]) is true,
+      a. Throw a TypeError exception.
+features: [class, class-methods-private]
+---*/
+
+let classStringExpression = `
+return class {
+  get m() { return 'test262'; }
+
+  access(o) {
+    return o.m;
+  }
+}
+`;
+
+let createAndInstantiateClass = function () {
+  return new (new (Function(classStringExpression)));
```

Comment:
This test is confusing me a little bit. We want to do this test using `Function` constructor, right? IIUC, `createAndInstantiateClass` will evaluate a new `ClassExpression` and then return an instantiation of this class. Is there any reason why you're using `new` to call `Function(classStringExpression)`? Also, I think it would be good if we construct the new function using `new operator`. What I have in mind is something like this:

```
let createAndInstantiateClass = function () {
    return new ((new Function(classStringExpression))());
}
```
or being more verbose, but making the code easier to follow, IMHO:
```
let createAndInstantiateClass = function () {
  let classFactoryFunction = new Function(classStringExpression);
  let unamedClass = classFactoryFunction();
	return new unamedClass();
}
```